Dockerizando aplicação

maikondouglas opened the pull request:
- Adicionado `Dockerfile.rails`
- Adicionado `docker-compose.yml`
- Adicionado `entryendpoints` para monitoramento do contêiner

maikon-douglas-vindi committed:
configure docker more compose

maikon-douglas-vindi committed:
Merge branch 'develop' of github.com:maikondouglas/one_account into feat/10-docker-compose-config

maikon-douglas-vindi committed:
add metrics for 132

maikon-douglas-vindi committed:
alter host database

lucas-carmichael-vindi commented on docker/Dockerfile.rails:
```diff
@@ -0,0 +1,17 @@
+ENTRYPOINT ["entrypoint.sh"]
+EXPOSE 3000
+
+CMD ["rails", "server", "-b", "0.0.0.0"]
```
Acho que esse cara pode ficar ali yml do compose, o que acha?  Ai vc removeria essa linha aqui :)

maikondouglas replied:
Pdc mano, faz sentido

lucas-carmichael-vindi commented on docker-compose.yml:
```diff
@@ -0,0 +1,24 @@
+      repository: git@github.com:maikondouglas/one_account.git
+    volumes:
+      - ./:/app
+    ports:
```
Por aqui você poderia por o comando `comands: "bin/rails server -b '0.0.0.0'"

lucas-carmichael-vindi replied:
https://github.com/maikondouglas/one_account/pull/12#discussion_r1343020293

maikondouglas replied:
Boa

lucas-carmichael-vindi commented on .rubocop_todo.yml:
```diff
@@ -134,7 +134,7 @@ Style/SymbolArray:
 # URISchemes: http, https
 Layout/LineLength:
-  Max: 120
+  Max: 132
```
qual o role de aumentar esse cara manin? duvida aqui :)

maikondouglas replied:
Na real, foi pra testar mas esse cara será removido e será mantido somente o `Metrics/LineLength:` que faz a mesma coisa que esse cara ai

lucas-carmichael-vindi commented on .rubocop_todo.yml:
```diff
@@ -151,7 +151,7 @@ Metrics/BlockLength:
 
 Metrics/LineLength:
-  Max: 80
+  Max: 132
```
https://github.com/maikondouglas/one_account/pull/12#discussion_r1343021996

maikondouglas replied:
https://github.com/maikondouglas/one_account/pull/12#discussion_r1349437926

lucas-carmichael-vindi commented on config/database.yml:
```diff
@@ -1,7 +1,7 @@
-  host: <%= ENV.fetch("DB_USERNAME") { 'localhost' } %>
+  host: <%= ENV.fetch("DB_HOST") { 'localhost' } %>
   username: <%= ENV.fetch("DB_USERNAME") { 'postgres' } %>
   password: <%= ENV.fetch("DB_PASSWORD") { 'postgres' } %>
```
vc esta usando o argumento, `POSTGRES_PASSWORD` e pq vc ta usando `DB_PASSWORD` ?

maikondouglas replied:
Puts, real mano... Estou usando o `DB_PASSWORD` lá no meu `.env`, vou mudar lá no compose

lucas-carmichael-vindi commented on docker/Dockerfile.rails:
```diff
@@ -0,0 +1,17 @@
+FROM ruby:3.2.0
+
+RUN apt-get update -qq && apt-get install -y build-essential libpq-dev nodejs postgresql-client
```
aqui vc poderia quebrar as linhas, o q acha?

```yml
RUN apt-get update -qq && \
         apt-get install -y build-essential libpq-dev nodejs postgresql-client
```

maikondouglas replied:
Boa mano, faz sentido sim... Como o Rubocop não questionou acabei deixando assim

lucas-carmichael-vindi commented on docker/Dockerfile.rails:
```diff
@@ -0,0 +1,17 @@
+FROM ruby:3.2.0
```
Poderíamos adicionar a configuração do usuário com uid 1000 e guid 1000 tbm o que acha? Assim vc esta executando o container como root e isso eh uma falha de segurança. 

O artigo eh um exemplo para talvez entender como pode fazer, https://caiodelgado.dev/entendendo-dockerfile/

maikondouglas replied:
Carai, bem pensando... Deixa-me ver aqui pq nunca fiz isso

maikon-douglas-vindi committed:
remove comand of dockerfile and add in compose

maikon-douglas-vindi committed:
- adjustment for use of environments
-  add entrypoint in bin/ and active method for verify state the pg
- create schema
- adjustment dockerfile for add group and user permission 1000

maikon-douglas-vindi committed:
Merge branch 'develop' of github.com:maikondouglas/one_account into feat/10-docker-compose-config
